xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -1552,23 +1552,21 @@ public Date getResourceLastModificationDate(String name)
 
     public byte[] getResourceContentAsBytes(String name) throws IOException
     {
-        InputStream is = null;
+        InputStream temp = null;
         if (getEngineContext() != null) {
 
             try {
-                is = getResourceAsStream(name);
+                temp = getResourceAsStream(name);
             } catch (Exception e) {
             }
         }
 
-        if (is == null) {
+        if (temp == null) {
             return FileUtils.readFileToByteArray(new File(name));
         }
 
-        try {
+        try (InputStream is = temp) {
```

Comment:
The previous code was not great in the first place but I would move this whole block and merge it with the content of `if (getEngineContext() != null) {` which would be a lot cleaner.